### PR TITLE
Remove ``AbstractLatentWorker.__init__``.

### PR DESCRIPTION
**Pull request by tomprince**
``AbstractLatentWorker`` is a ``BuildbotService``, which has an
``__init__`` that delegates to ``checkConfig``. ``AbstractLatentWorker``
should just have an ``checkConfig`` and not override
``BuildbotService.__init__``.

### CODE REVIEW

File: master/buildbot/worker/base.py [hunk 1]
```diff
@@ -77,6 +79,8 @@ def checkConfig(self, name, password, max_builds=None,
         @type locks: dictionary
         """
         self.name = name = ascii2unicode(name)
+        self.build_wait_timeout = build_wait_timeout
```
Comment (tardyp) [inline master/buildbot/worker/base.py]: you rather need to update this in reconfigureService(), or the reconfiguration will not work if build_wait_timeout is changed (better than nothing, though)

**Comment by tardyp** [comment]
also, the subclasses are also implementing __init__ (just checked docker)

File: master/buildbot/worker/base.py (hunk 1 above)
Comment (tomprince) [inline master/buildbot/worker/base.py]: Thanks for catching this. It appears that this previously wan't being reconfigured before.

Comment (tardyp) [inline master/buildbot/worker/base.py]: indeed, there are still a bunch of services that needs to be migrating to the reconfigure api.
schedulers and workers are one of those. 
This means buildbot reload won't apply changes for those.

I need to make the list, and see if we want to release with our without the fix.
not complicated, but requires time to make sure this is properly tested.

**Comment by rutsky** [comment]
Windows build fails because of errors during `lz4` installation, looks like this bug: https://github.com/darkdragn/lz4tools/issues/6

File: master/buildbot/test/unit/test_worker_base.py [hunk 2]
```diff
@@ -471,6 +471,37 @@ def test_shutdownRequested(self):
...
+        self.clock = task.Clock()
+        self.patch(reactor, 'callLater', self.clock.callLater)
+        self.patch(reactor, 'seconds', self.clock.seconds)
+
+    def do_test_reconfigService(self, old, new, existingRegistration=True):
+        old.parent = self.master
+        if existingRegistration:
+            old.registration = bworkermanager.FakeWorkerRegistration(old)
+        old.missing_timer = mock.Mock(name='missing_timer')
+        self.successResultOf(old.startService())
```
Comment (tardyp) [inline master/buildbot/test/unit/test_worker_base.py]: I wasn't aware of this method.

This fails with twisted 12:

  File "/home/travis/build/buildbot/buildbot/master/buildbot/test/unit/test_worker_base.py", line 500, in test_reconfigService
    self.do_test_reconfigService(old, new)
  File "/home/travis/build/buildbot/buildbot/master/buildbot/test/unit/test_worker_base.py", line 492, in do_test_reconfigService
    self.successResultOf(old.startService())
exceptions.AttributeError: 'TestAbstractLatentWorker' object has no attribute 'successResultOf'

Looks like was introduced in 13.
I am not against dropping such old version of twisted, but we should probably ask the ml first.

**Comment by tomprince** [comment]
> also, the subclasses are also implementing init (just checked docker)

That should be changed, but I don't think that is a prerequisite for this fix? One trick with updating the actual latent workers is that if the credentials change, we won't be able to shutdown any currently running instances.

File: master/buildbot/test/unit/test_worker_base.py (hunk 2 above)
Comment (tomprince) [inline master/buildbot/test/unit/test_worker_base.py]: We have now dropped twisted < 14.0.

**Comment by tardyp** [comment]
👍

**Pull request merged**